src/sentry/spans/consumers/process/factory.py

```diff
@@ -158,12 +155,34 @@ def _format_event_id(payload: Mapping[str, Any]) -> str:
     return ""
 
 
+def _deserialize_payload(payload: bytes) -> Mapping[str, Any]:
+    try:
+        return msgpack.unpackb(payload)
+    except msgpack.FormatError:
+        return json.loads(payload, use_rapid_json=True)
```

Comment:
Can we add a comment explaining this is for backwards compatibility between v1 and v2?